docs/references/fluxyaml-config-files.md

```diff
@@ -1,229 +1,268 @@
-# Manifest factorization through `.flux.yaml` configuration files
+# Manifest generation through `.flux.yaml` configuration files
 
-## Enabling search of `.flux.yaml` files
+This feature lets you generate Kubernetes manifests with a program,
+instead of having to include them in your git repo as YAML files. For
+example, you can use `kustomize` to patch a common set of resources to
+suit a particular environment.
 
-To enable it supply the command-line flag `--manifest-generation=true` to `fluxd`.
+## Setting manifest generation up
 
-## Goal
+Manifest generation is controlled by the flags given to `fluxd`, and
+`.flux.yaml` files in your git repo.
 
-It is a common pattern to run very similar resources in separate clusters. There are various scenarios in which this is
-required, the two main ones being:
+To enable it, pass the command-line flag `--manifest-generation=true`
```

Comment:
Maybe we should highlight this a bit more, since it's essential for the feature. Do we use a highlight box or something like that in other parts of the docs?

Reply:
I moved it to the top and made the absolute requirements clearer.